aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -928,6 +1097,49 @@ void convert_indices_from_csr_to_coo_cpu(
         }
       });
 }
+
+// This function unfolds the compressed indices of a compressed sparse matrix
+// into a batched compressed sparse tensor.
+// This is analogous to an unflatten-like operation:
+// unflatten(0, {b, r}) for csr/bsr with input shape (r*b, c)
+//          (output shape (b, r, c))
+// unflatten(1, {b, c}).transpose(0,1) for csc/bsc with input shape (r, c*b)
+//          (output shape (r, b, c) unflatten, (b, r, c) unflatten + transpose)
```

Comment:
I am not convinced we need a kernel for that. You can create a buffer and use a combination of view/transpose and copy_ to accomplish this.